components/omega/doc/design/EOS.md

```diff
@@ -0,0 +1,158 @@
+computing the specific volume, and its first derivatives.
+
+## 2 Requirements
+
+### 2.1 Requirement: calculate specific volume/density
+The specific volume and density will be calculated form from prognostic
+temperature, salinity, and pressure (ref function `gsw_specvol(sa,ct,p`)).
+
+### 2.2 Requirement: allow a choice of equations of state
+Omega-1 should include the option for two EOS options. The first is a linear
+eos, for benchmarking against MPAS-O and to use in simplified cases. The other
+option should be a well-established version of TEOS-10 (e.g. Roquet 75-term
+polynomial). 
+
+### 2.3 Requirement: computational efficiency
+The eos code should not be major computational bottleneck (criteria?)
+
+### 2.4 Requirement: abide by license for the GSW Oceanographic Toolbox
+The GSW toolbox can only be used without modification, so a Omega-specific port
+of certain functions will be required to support Kokkos/GPU enabled
+computation.  The toolbox without modifications can still be used for testing.
+
+### 2.5 Requirement: verification 
+The Omega implementation should be compared with the test value published in
+Roquet et al 2015, and the official GSW Oceanographic Toolbox (supported by
+TEOS-10) to verify correctness.
+
+### 2.6 Desired: freezing temperature calculation
+Later versions should include calculating the freezing temperature of seawater.
+
+### 2.7 Desired: output in situ temperature
+Later versions should include calculating the in situ  temperature needed for
+coupling (at surface including at non-zero pressure).
+
+### 2.8 Desired: output first derivatives
+The EOS will need to be able to produce alpha and beta (drho/dT and drho/dS)
+for linear expansions in the higher-order pressure gradient.
+
+### 2.9 Desired: check value range
+Later versions should include a check on the range of ocean properties to
+assess use of TEOS-10 (equivalent to the ``ocean funnel'' check).
+
+### 2.10 Desired: check value range
+Later versions should include functions to convert between conservative and
+potential temperatures, and absolute and practical salinity. These will be used
+to convert the initial conditions or compare to ocean states from other sources
+(MPAS-Ocean, reanalysis, other models etc.) These functions may be used offline
+in pre- or post-processing but need to be consistent with the EOS
+implementation. 
+
+## 3 Algorithmic Formulation
+
+## 4 Design
+The GSW-C toolbox will be incorporated into Omega as a submodule.
+In order to abide by the license of the toolbox, the toolbox will only be used
+in testing to check our implementation. The coefficients for the Roquet et al.
+2015 expansion will be based on the published values (Appendix). A Kokkos
+implementation of the function to compute specific volume will be ported to
+Omega. The GSW-C toolbox sub-module will serve as a baseline reference for our
+ports in unit tests. 
+
+For flexibility in optimizing performance, calls to the EOS function should be
```

Comment:
As mentioned above, I suspect we want to restrict computing density to one time per step and given the EOS complexity, probably won't be inlined or merged with other kernels. Guess we shouldn't design it out, but the preference would be whole-array if there are any compromises associated with the single-cell support.